internal/compliance/snapshot_poller/pollers/aws/poll.go

```
@@ -173,6 +174,38 @@ func Poll(scanRequest *pollermodels.ScanEntry) (
 		NextPageToken: scanRequest.NextPageToken,
 	}
 
+	// Retrieve source integration from a short-lived cache for integration filtering options
+	var sourceIntegration *models.SourceIntegration
+	if scanRequest.IntegrationID == nil {
+		zap.L().Error("got a malformed scan request - no integration id", zap.Any("scan request", scanRequest))
+		return nil, nil
+	}
+	sourceIntegration, err = utils.GetIntegration(*scanRequest.IntegrationID)
+	if err != nil {
+		zap.L().Warn("failed to retrieve integration from source api")
```

Why log a warning? Why not return an error?

---

Yeah I think returning an error here is probably better, most likely case this would happen:

1. Transient network error meaning communication with the `source-api` is bad
2. The user deleted the source

The second case might be a bit annoying to see a bunch of errors for, but we would also see a bunch of errors if we tried to scan and they had deleted the audit role (and presumably if they delete the source they would prefer not  to have a bunch of inflight messages continue scanning).